src/darwin/Framework/CHIP/MTRDeviceController_Concrete.mm

```diff
@@ -450,6 +450,7 @@ - (void)cleanupAfterStartup
         [device invalidate];
     }
     [self stopBrowseForCommissionables];
+    [self clearDeviceControllerDelegates];
```

Comment:
Could we just put this in the MTRDeviceController shutdown (and make sure we call it from the subclass)?

In practice, since there can't be any delegates until startup completes successfully, doing things in shutdown instead of here should be ok.  Things need to go in this method if they might need cleanup after a partial startup, when the consumer does not actually have a controller to call shutdown on.

Reply:
Done!